src/Process/PointCloudProcessing.js

```diff
@@ -212,78 +143,86 @@ export default {
     },
 
     update(context, layer, elt) {
-        const { shouldBeLoaded } = shouldDisplayNode(context, layer, elt);
+        elt.visible = false;
 
-        elt.shouldBeLoaded = shouldBeLoaded;
+        if (layer.octreeDepthLimit >= 0 && layer.octreeDepthLimit < elt.name.length) {
+            markForDeletion(elt);
+            return;
+        }
+
+        // pick the best bounding box
+        const bbox = (elt.tightbbox ? elt.tightbbox : elt.bbox);
+        elt.visible = context.camera.isBox3Visible(bbox, layer.object3d.matrixWorld);
+        if (!elt.visible) {
+            markForDeletion(elt);
+            return;
+        }
 
-        if (shouldBeLoaded > 0) {
-            elt.notVisibleSince = undefined;
+        elt.notVisibleSince = undefined;
 
-            // only load geometry if this elements has points
-            if (elt.numPoints > 0) {
-                if (elt.obj) {
-                    elt.obj.material.visible = true;
-                    if (__DEBUG__) {
-                        elt.obj.material.uniforms.density.value = elt.density;
+        // only load geometry if this elements has points
+        if (elt.numPoints > 0) {
+            if (elt.obj) {
+                elt.obj.material.visible = true;
+                elt.obj.material.uniforms.size.value = layer.pointSize;
 
-                        if (layer.bboxes.visible) {
-                            if (!elt.obj.boxHelper) {
-                                initBoundingBox(elt, layer);
-                            }
-                            elt.obj.boxHelper.visible = true;
-                            elt.obj.boxHelper.material.color.r = 1 - shouldBeLoaded;
-                            elt.obj.boxHelper.material.color.g = shouldBeLoaded;
+                if (__DEBUG__) {
+                    if (layer.bboxes.visible) {
+                        if (!elt.obj.boxHelper) {
+                            initBoundingBox(elt, layer);
                         }
+                        elt.obj.boxHelper.visible = true;
+                        elt.obj.boxHelper.material.color.r = 1 - elt.sse;
+                        elt.obj.boxHelper.material.color.g = elt.sse;
+                    }
+                }
+            } else if (!elt.promise) {
+                const distance = Math.max(0.001, bbox.distanceToPoint(context.camera.camera3D.position));
+                // Increase priority of nearest node
+                const priority = computeScreenSpaceError(context, layer, elt, distance) / distance;
```

Comment:
I could but that would complexify a bit the code at line 127 (with a `if (elt.sse is already computed)`) for something that only happens once per node: when we decide to fetch them.

Reply:
Would it be guaranteed to be computed on line 127? (because elt.promise would be undefined the first time update is called on this node). But even if so, I agree this is a bit fragile. We can keep it like that.